src/main/cpp/src/get_json_object.cu

```diff
@@ -294,26 +319,24 @@ __launch_bounds__(block_size) CUDF_KERNEL
   }
 }
 
-std::unique_ptr<cudf::column> get_json_object(cudf::strings_column_view const& col,
-                                              cudf::string_scalar const& json_path,
-                                              spark_rapids_jni::json_parser_options options,
-                                              rmm::cuda_stream_view stream,
-                                              rmm::mr::device_memory_resource* mr)
+std::unique_ptr<cudf::column> get_json_object(
+  cudf::strings_column_view const& col,
+  std::vector<std::tuple<path_instruction_type, std::string, int64_t>> const& instructions,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr)
 {
   if (col.is_empty()) return cudf::make_empty_column(cudf::type_id::STRING);
 
-  // parse the json_path into a command buffer
-  auto path_commands_optional = parse_path(json_path);
-
-  // if the json path is empty, return a string column containing all nulls
-  if (!path_commands_optional.has_value()) {
-    return std::make_unique<cudf::column>(
-      cudf::data_type{cudf::type_id::STRING},
-      col.size(),
-      rmm::device_buffer{0, stream, mr},  // no data
-      cudf::detail::create_null_mask(col.size(), cudf::mask_state::ALL_NULL, stream, mr),
-      col.size());                        // null count
+  // get a string buffer to store all the names and convert to device
+  std::string all_names;
+  for (auto const& inst : instructions) {
+    all_names += std::get<1>(inst);
   }
+  cudf::string_scalar all_names_scalar(all_names);
```

Comment:
Please be careful the life cycle of this class.
It's local variable, and the device memory in the scalar will be de deallocated soon.
Please double check.

Reply:
Sorry for this, it's not a problem.